### PR TITLE
added crux-connect-protocol message schema and testcase for the same

### PR DESCRIPTION
**Pull request by indramalav**

### CODE REVIEW

File: src/infrastructure/implementations/crux-messenger.ts [hunk 1]
```diff
@@ -117,3 +117,14 @@ export const cruxPaymentProtocol: IMessageSchema[] = [{
             .required().min(36).max(36),
     }),
 }];
+
+export const cruxConenctProtocol: IMessageSchema[] = [{
```
Comment (quixote911) [inline src/infrastructure/implementations/crux-messenger.ts]: fix spelling

File: src/infrastructure/implementations/crux-messenger.ts [hunk 2]
```diff
@@ -117,3 +117,14 @@ export const cruxPaymentProtocol: IMessageSchema[] = [{
...
+    messageType: "CONNECT_REQUEST",
+    schema: Joi.object({
+        publicKey: Joi.string()
+            .required(),
+
+        cruxId: Joi.string()
```
Comment (quixote911) [inline src/infrastructure/implementations/crux-messenger.ts]: Whose cruxId is this?

File: src/infrastructure/implementations/crux-messenger.ts [hunk 3]
```diff
@@ -117,3 +117,14 @@ export const cruxPaymentProtocol: IMessageSchema[] = [{
             .required().min(36).max(36),
     }),
 }];
+
+export const cruxConenctProtocol: IMessageSchema[] = [{
+    messageType: "CONNECT_REQUEST",
+    schema: Joi.object({
+        publicKey: Joi.string()
+            .required(),
```
Comment (quixote911) [inline src/infrastructure/implementations/crux-messenger.ts]: Fix length of public key. look at PAYMENT_REQUEST

File: src/test/crux-messenger/test-connect-protocol.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                cruxId: "yadu007@cruxdev.crux"
+            }];
+            for (let msg of validConnectRequests) {
+                const result = this.messageSchema.validate(msg)
+                console.log(result);
+                expect(result.error).to.be.undefined;
+            }
+        });
+
+        it('Invalid Messages', async function() {
```
Comment (quixote911) [inline src/test/crux-messenger/test-connect-protocol.ts:35]: Add test case regarding invalid public key due to length.

**Review by quixote911 — CHANGES_REQUESTED** [review]

File: src/test/crux-messenger/test-connect-protocol.ts (hunk 4 above)
Comment (indramalav) [reply src/test/crux-messenger/test-connect-protocol.ts:35]: doing

File: src/infrastructure/implementations/crux-messenger.ts (hunk 3 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: sure allowed max length??

File: src/infrastructure/implementations/crux-messenger.ts (hunk 2 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: will do it

File: src/infrastructure/implementations/crux-messenger.ts (hunk 1 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: yaar ye phir se ho gyi mistake

File: src/infrastructure/implementations/crux-messenger.ts (hunk 2 above)
Comment (quixote911) [reply src/infrastructure/implementations/crux-messenger.ts]: but actually kiski crux id hai ye? Sender has no Crux ID. Recipient has Crux ID but why does it need to be there in this message?

File: src/infrastructure/implementations/crux-messenger.ts (hunk 3 above)
Comment (quixote911) [reply src/infrastructure/implementations/crux-messenger.ts]: Max and Min length is fixed - just count length of any valid public key

Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: okay cool

File: src/infrastructure/implementations/crux-messenger.ts (hunk 2 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: actually its not required, mene ye disapprove wale case ke lie daal di thi hata deta hoon

File: src/test/crux-messenger/test-connect-protocol.ts (hunk 4 above)
Comment (indramalav) [reply src/test/crux-messenger/test-connect-protocol.ts:35]: done

File: src/infrastructure/implementations/crux-messenger.ts (hunk 3 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: done

File: src/infrastructure/implementations/crux-messenger.ts (hunk 2 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: removed

File: src/infrastructure/implementations/crux-messenger.ts (hunk 1 above)
Comment (indramalav) [reply src/infrastructure/implementations/crux-messenger.ts]: done

**Comment by quixote911** [comment]
Closing this because we no longer have any separate CONNECT protocol.

**Pull request closed without merging**